### PR TITLE
Improve tab implementation

### PR DESCRIPTION
**Pull request by oshorefueled**
### Resolves

Issue #73

### What's new

This PR improves the current tab implementation by 
*  adding an indicator and and an icon to the tab
*  adding a position option to display the tab on the 4 sides of a page.

### Relevant screenshots or logs
Screenshot 2020-04-27 at 3 05 41 PM

### CODE REVIEW

File: ui/decredmaterial/tabs.go [hunk 1]
```diff
@@ -1,71 +1,238 @@
...
+	default:
+		t.list.Axis = layout.Vertical
+		t.Flex.Axis = layout.Horizontal
+	}
+
+	widgets := t.contentTabPosition(gtx, body)
+	t.Flex.Spacing = layout.SpaceBetween
+	t.Flex.Layout(gtx, widgets...)
+	for i := range t.btns {
+		t.changed = false
```
Comment (oluwandabira) [inline ui/decredmaterial/tabs.go]: This should be outside the for. I'm not sure why it worked before

File: ui/decredmaterial/tabs.go [hunk 2]
```diff
@@ -1,71 +1,238 @@
...
+	case Top, Bottom:
+		t.list.Axis = layout.Horizontal
+		t.Flex.Axis = layout.Vertical
+	default:
+		t.list.Axis = layout.Vertical
+		t.Flex.Axis = layout.Horizontal
+	}
+
+	widgets := t.contentTabPosition(gtx, body)
+	t.Flex.Spacing = layout.SpaceBetween
```
Comment (oluwandabira) [inline ui/decredmaterial/tabs.go]: Unexport the `Flex` and set the spacing during the `Tabs` creation

File: ui/decredmaterial/tabs.go [hunk 3]
```diff
@@ -1,71 +1,238 @@
...
 }
 
-func (tab *Tabs) Changed() bool {
-	return tab.changed
+func (t *Tabs) Changed() bool {
+	return t.changed
+}
+
+// contentTabPosition depending on the specified tab position determines the order of the tab and
+// the page content.
```
Comment (oluwandabira) [inline ui/decredmaterial/tabs.go]: Could you reword this?

File: ui/decredmaterial/tabs.go [hunk 4]
```diff
@@ -1,71 +1,238 @@
...
+			}
+			indicatorDirection(tabPosition).Layout(gtx, func() {
+				layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+					layout.Rigid(activeIndicator(gtx, tabWidth, tabHeight)))
+			})
+		}),
+	)
+}
+
+// Tabs lays out a Flexed(Size) List with Selected as the first element and Item as the rest.
```
Comment (oluwandabira) [inline ui/decredmaterial/tabs.go]: Change this.

File: ui/decredmaterial/tabs.go [hunk 5]
```diff
@@ -1,71 +1,238 @@
...
+	default:
+		width, height = 5, gtx.Dimensions.Size.Y
+	}
+	return
 }
 
-// Tabs laysout a Flexed(Size) List with Selected as the first element and Item as the rest.
+// tabAlignment determines the alignment of the active tab indicator relative to the tab item
+// content. It is determined by the position of the tab.
+func indicatorDirection(tabPosition Position) layout.Direction {
```
Comment (oluwandabira) [inline ui/decredmaterial/tabs.go]: Wrong comment.

File: ui/decredmaterial/tabs.go [hunk 6]
```diff
@@ -1,71 +1,238 @@
...
+	case Bottom:
+		return layout.N
+	case Right:
+		return layout.E
+	default:
+		return layout.W
+	}
+}
+
+// indicator defines how the active tab indicator is drawn
```
Comment (oluwandabira) [inline ui/decredmaterial/tabs.go]: Wrong comment.

**Review by oluwandabira — CHANGES_REQUESTED** [review]

File: ui/decredmaterial/tabs.go [hunk 7]
```diff
@@ -1,71 +1,306 @@
...
-	)
-	for i := range tab.btns {
-		tab.changed = false
-		if tab.btns[i].Clicked(gtx) {
-			tab.changed = true
-			tab.Selected = i
+// scrollButton lays out the right and left scroll buttons of the tab when Position is Horizontal.
+func (t *Tabs) scrollButton(gtx *layout.Context, right bool, button *widget.Button) layout.FlexChild {
+	show := false
+	icon := mustIcon(NewIcon(icons.NavigationChevronLeft))
```
Comment (oluwandabira) [inline ui/decredmaterial/tabs.go:207]: Decoding the icon should be done before the icon needs to be laid out and done only once.

File: ui/decredmaterial/tabs.go [hunk 8]
```diff
@@ -1,71 +1,306 @@
...
+	Bottom
+	Left
+)
 
+var adaptiveTabWidth int
+
+// Position determines what side of the page the tab would be laid out
+type Position int
+
+// TabItem displays a single child of a tab. Label and Icon in TabItem are optional.
```
Comment (oluwandabira) [inline ui/decredmaterial/tabs.go:33]: If both `Label` and `Icon` are optional, what does it display when neither is present.

**Review by oluwandabira — CHANGES_REQUESTED** [review]

File: ui/decredmaterial/tabs.go (hunk 8 above)
Comment (oshorefueled) [reply ui/decredmaterial/tabs.go:33]: Just tab items with the minimum width of the active indicator is displayed.

**Review by oluwandabira — APPROVED** [review]

**Pull request merged**